doc/environ.sh.sample

```diff
@@ -1,102 +1,173 @@
 export DC_ARM_LDFLAGS=""
 
-# Setup some default CFLAGS for compilation. The things that will go here
-# are user specifyable, like optimization level and whether you want stack
-# traces enabled. Some platforms may have optimization restrictions,
-# please check README.
-# GCC seems to have made -fomit-frame-pointer the default on many targets, so
-# hence you may need -fno-omit-frame-pointer to actually have GCC spit out frame
-# pointers. It won't hurt to have it in there either way.
-# Link-time optimizations can be enabled by adding -flto. It however requires a
-# recent toolchain (GCC 10+), and has not been thoroughly tested.
-export KOS_CFLAGS="-O2 -fomit-frame-pointer"
-# export KOS_CFLAGS="-O2 -DFRAME_POINTERS -fno-omit-frame-pointer"
+# Optimization Level 
+#
+# Controls the baseline optimization level to use when building. 
+# Typically this is -Og (debugging), -O0, -01, -02, or -03.
+# NOTE: For our target, -O4 is a valid optimization level that has 
+# been seen to have some performance impact as well.
+#
+export KOS_CFLAGS="${KOS_CFLAGS} -O2"
+
+# Additional Optimizations
+#
+# Uncomment this to enable what has been found emperically to be 
+# the optimal set of additional flags for release build performance 
+# on the current stable toolchain.
+#
+#export KOS_CFLAGS="${KOS_CFLAGS} -freorder-blocks-algorithm=simple -flto=auto"
+
+# Frame Pointers
+#
+# Controls whether frame pointers are emitted or not. Disabled by
+# default. Enable them if you plan to use GDB for debugging. 
+#
+export KOS_CFLAGS="${KOS_CFLAGS} -fomit-frame-pointer"
+#export KOS_CFLAGS="${KOS_CFLAGS} -fno-omit-frame-pointer -DFRAME_POINTERS"
 
+# GCC Builtin Functions
+#
 # Comment out this line to enable GCC to use its own builtin implementations of 
 # certain standard library functions. Under certain conditions, this can allow
 # compiler-optimized implementations to replace standard function invocations.
 # The downside of this is that it COULD interfere with Newlib or KOS implementations
 # of these functions, and it has not been tested thoroughly to ensure compatibility. 
+#
 export KOS_CFLAGS="${KOS_CFLAGS} -fno-builtin"
 
+# Fast Math Instructions
+#
 # Uncomment this line to enable the optimized fast-math instructions (FSSRA,
 # FSCA, and FSQRT) for calculating sin/cos, inverse square root, and square roots.
 # These can result in substantial performance gains for these kinds of operations;
 # however, they do so at the price of accuracy and are not IEEE compliant.
 # NOTE: This also requires -fno-builtin be removed from KOS_CFLAGS to take effect!
+#
 # export KOS_CFLAGS="${KOS_CFLAGS} -ffast-math -ffp-contract=fast -mfsrra -mfsca"
 
-# Everything else is pretty much shared. If you want to configure compiler
-# options or other such things, look at this file.
+# Additional Tools Path
+#
+# If not already set, add "bin" directory to PATH variable, which is where
+# addititional tools such as dc-tool-ip and dc-tool-serial are typically 
```

Comment:
Typo: addititional => additional

Reply:
Thank you so much for catching my dumb typos. Can't have them in such an important file as one of the first things the user sees. :)